src/metabase/driver/sql_jdbc/sync/describe_database.clj

```diff
@@ -107,44 +106,77 @@
                           :schema      (.getString rset "TABLE_SCHEM")
                           :description (when-let [remarks (.getString rset "REMARKS")]
                                          (when-not (str/blank? remarks)
-                                           remarks))}))))))
+                                           remarks))
+                          :type        (.getString rset "TABLE_TYPE")}))))))
```

Comment:
`current-user-table-privileges` does not return privileges for external tables on redshift, and foreign table on postgres, so we need to use the select method on them, so to check select privilege for these tables we need to use the old method of trying a `select` query. In order to do that we need to return the type so we know what's the type of table we're dealing with.

see line 134.